app/src/lib/stores/commit-status-store.ts

```diff
@@ -331,11 +316,56 @@ export class CommitStatusStore {
       checks.push(...latestCheckRunsByName.map(apiCheckRunToRefCheck))
     }
 
-    const check = createCombinedCheckFromChecks(checks)
+    let checksWithActions = null
+    if (subscription.branchName !== undefined) {
+      checksWithActions = await this.getAndMapActionWorkflowRunsToCheckRuns(
+        checks,
+        key,
+        subscription.branchName
+      )
+    }
+
+    const check = createCombinedCheckFromChecks(checksWithActions ?? checks)
     this.cache.set(key, { check, fetchedAt: new Date() })
     subscription.callbacks.forEach(cb => cb(check))
   }
 
+  private async getAndMapActionWorkflowRunsToCheckRuns(
+    checks: ReadonlyArray<IRefCheck>,
+    key: string,
+    branchName: string
+  ): Promise<ReadonlyArray<IRefCheck> | null> {
+    const existingChecks = this.cache.get(key)
+
+    // If the checks haven't changed since last status refresh, don't bother
+    // retrieving actions workflows and they could be stale if this is directly after a rerun
+    if (
+      existingChecks !== undefined &&
+      existingChecks.check !== null &&
+      existingChecks.check.checks.find(c => c.actionsWorkflow !== undefined) !==
+        undefined &&
```

Comment:
I think this should do the same, no?
```suggestion
      existingChecks.check.checks.some(c => c.actionsWorkflow !== undefined) &&
```

Reply:
Good call - I always forget about `some`